addon/components/flexberry-maptoolbar.js

```diff
@@ -30,63 +98,163 @@ export default Ember.Component.extend({
+      @param {BaseMapCommand} e.mapCommand Map command that must be executed.
+      @param {<a href="http://learn.jquery.com/using-jquery-core/jquery-object/">jQuery-object</a>} e.target
+      jQuery element related to map command that must be executed.
+      @param {<a href="http://learn.jquery.com/events/introduction-to-events/#the-event-object">jQuery event object</a>} e.originalEvent
+      Event object related to event that triggered this action.
     */
-    callCommand(commandName, ...args) {
-      let command = getOwner(this).lookup('mapcommand:' + commandName);
-      let leafletMap = this.get('leafletMap');
-      if (command && leafletMap) {
-        command.execute(leafletMap, ...args);
+    onMapCommandExecute(...args) {
+      // Actions event object is always the last parameter.
+      let e = args[args.length - 1] || {};
+      let mapCommand = Ember.get(e, 'mapCommand');
+      Ember.assert(
+        `Wrong type of \`e.mapCommand\` parameter: actual type is ${Ember.typeOf(mapCommand)}, but 'map-command' is expected.`,
+        mapCommand instanceof BaseMapCommand);
+
+      if (Ember.isNone(Ember.get(mapCommand, 'leafletMap'))) {
+        Ember.set(mapCommand, 'leafletMap', this.get('leafletMap'));
+      }
+
+      if (Ember.isNone(Ember.get(mapCommand, 'layers'))) {
+        Ember.set(mapCommand, 'layers', this.get('layers'));
       }
+
+      mapCommand.execute(...args);
+    }
+  },
+
+  /**
+    Activates default map-tool.
+
+    @method _activateDefaultMapTool
+    @private
+  */
+  _activateDefaultMapTool() {
+    let defaultMapToolName = this.get('defaultMapToolName');
+    Ember.assert(
+      `Required \'defaultMapToolName\' property is not defined in flexberry-maptoolbar (${this.toString()})`,
+      !Ember.isBlank(defaultMapToolName));
+
+    let getDefaultMapToolComponent = function(root) {
+      let defaultMapToolComponent = null;
+      Ember.A(root.get('childViews') || []).forEach((child) => {
+        let component = child instanceof BaseMapToolComponent ? child : getDefaultMapToolComponent(child);
+        if (component instanceof BaseMapToolComponent && Ember.get(component, 'name') === defaultMapToolName) {
+          defaultMapToolComponent = component;
+          return false;
+        }
+      });
+
+      return defaultMapToolComponent;
+    };
+
+    let defaultMapToolComponent = getDefaultMapToolComponent(this);
+    Ember.assert(
+      `Map-tool component related to \`${defaultMapToolName}\` map-tool doesn\`t exist in flexberry-maptoolbar (${this.toString()})`,
+      !Ember.isNone(defaultMapToolComponent));
+
+    defaultMapToolComponent.activateMapTool();
+  },
+
+  /**
+    Observes changes in {{#crossLink "FlexberryMaptoolbarComponent/leafletMap:propery"}}'leafletMap' property{{/crossLink}}.
+    Activates default map-tool when leafletMap initialized.
+
+    @method _leafletMapDidChange
+    @private
+  */
+  _leafletMapDidChange: Ember.on('init', Ember.observer('leafletMap', function() {
```

Comment:
Because same logic must be executed on 'init', if leafletMap is defined in initialization time.